napari/layers/labels/labels.py

```diff
@@ -465,9 +516,20 @@ def _raw_to_displayed(self, raw):
         image : array
             Image mapped between 0 and 1 to be displayed.
         """
-        image = np.where(
-            raw > 0, colormaps._low_discrepancy_image(raw, self._seed), 0
-        )
+        if self.enable_custom_color_dict:
+            u, inv = np.unique(raw, return_inverse=True)
```

Comment:
I think here we need to have some discussion around what we want the behavior to be for labels outside of the color_dict. Options are

1) Make them same as background (i.e. transparent)

2) Throw an error - painting outside of dict not supported

3) Have some other default color they get painted with

4) Use the color they would have had with the random color map

5) Do some mathematics to make the label "appear" inside the color_dict (i.e. treat like a cycle or something).

Unfortunately it's not clear to me what the right behavior should be with the dict approach, but I think overall we should aim for a high performance approach.

I think I like the idea of emitting a warning and using the background color right now, but I'm open to other ideas

Reply:
I think we do need to somehow give the pixels with labels not in the custom colormap a non-transparent (usually background) color. This is because in the GUI, the user can select a label and start painting and they should see something. I suppose the alternative is the restrict the label picker to things in the color map.

Reply:
how about we map non found labels to black? Seems like a reasonable first choice for now - you see something, but it's maybe not what you want? Can improve in the future.

Reply:
right now I am using a random value in the same colormap, the problem with black is that it does not exist in the given colormap, maybe we always add black in the custom color map?

Reply:
I tried the black approach, it looks fine to me, gonna push it soon